src/bootstrap/tool.rs

```diff
@@ -162,7 +162,8 @@ impl Step for ToolBuild {
                 "the following dependencies are duplicated although they \
                       have the same features enabled:"
             );
-            for (id, cur, prev) in duplicates.drain_filter(|(_, cur, prev)| cur.2 == prev.2) {
+            duplicates.retain(|(_, cur, prev)| cur.2 != prev.2);
+            for (id, cur, prev) in &duplicates {
```

Comment:
This doesn't seem to have the same behavior - it prints the duplicates with the same features both here and below. Maybe use `partition` instead?
```suggestion
            let (same_features, different_features) = duplicates.partition(|(_, cur, prev)| cur.2 == prev.2);
            for (id, cur, prev) in same_features {
```
and then change line 172 to say `in different_features`.

Reply:
Fixed, thanks! I missed that drain_filter returns the drained elements as part of the iterator :/

We should probably add the partition suggestion to the drain_filter "alternatives" list.